price-feeder/oracle/oracle.go

```diff
@@ -179,10 +190,28 @@ func (o *Oracle) SetPrices() error {
 	}
 
 	if err := g.Wait(); err != nil {
+		o.logger.Debug().Err(err)
+		return nil
+	}
+
+	reportedRates := []string{}
+	for _, providers := range providerPrices {
+		for base := range providers {
+			if !Contains(reportedRates, base) {
+				reportedRates = append(reportedRates, base)
+			}
+		}
+	}
+	if len(reportedRates) != len(requiredRates) {
+		return fmt.Errorf("unable to get prices for all exchange rates")
+	}
```

Comment:
I think this should do it! But I wonder if we're better off just using a map instead of a slice.

1. Ensure they're the same length
2. Ensure all keys in required exist in provided